distributed/security.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import print_function, division, absolute_import
+
+import ssl
+
+from . import config
+
+
+_roles = ['client', 'scheduler', 'worker']
+
+_tls_per_role_fields = ['key', 'cert']
+
+_tls_fields = ['ca_file', 'ciphers']
+
+_misc_fields = ['require_encryption']
+
+_fields = set(_misc_fields +
+              ['tls_%s' % field for field in _tls_fields] +
+              ['tls_%s_%s' % (role, field)
+               for role in _roles
+               for field in _tls_per_role_fields]
+              )
+
+
+def _field_to_config_key(field):
+    return field.replace('_', '-')
+
+
+class Security(object):
+    """
+    An object to gather and pass around security configuration.
+    Default values are gathered from the global ``config`` object and
+    can be overriden by constructor args.
+
+    Supported fields:
+        - require_encryption
+        - tls_ca_file
+        - tls_ciphers
+        - tls_client_key
+        - tls_client_cert
+        - tls_scheduler_key
+        - tls_scheduler_cert
+        - tls_worker_key
+        - tls_worker_cert
+    """
```

Comment:
Is it likely that we will be asked to support other security systems in the future other than TLS?  If so then is the API that we're setting up here going to cause problems if we have to grow in other direction?